lib/indexer/indexer.js

```diff
@@ -0,0 +1,689 @@
+/*!
+ * indexer.js - storage for indexes
+ * Copyright (c) 2018, the bcoin developers (MIT License).
+ * https://github.com/bcoin-org/bcoin
+ */
+
+'use strict';
+
+const assert = require('assert');
+const path = require('path');
+const fs = require('bfile');
+const bio = require('bufio');
+const EventEmitter = require('events');
+const Logger = require('blgr');
+const Network = require('../protocol/network');
+const util = require('../utils/util');
+const layout = require('./layout');
+const CoinView = require('../coins/coinview');
+const Block = require('../primitives/block');
+const {ZERO_HASH} = require('../protocol/consensus');
+
+/**
+ * Indexer
+ * @alias module:indexer.Indexer
+ * @extends EventEmitter
+ * @property {IndexerDB} db
+ * @property {Number} height
+ * @emits Indexer#chain tip
```

Comment:
nit: outdated jsdoc, probably just remove everything and mention abstract class.